streams/src/test/java/org/apache/kafka/streams/utils/UniqueTopicSerdeScope.java

```diff
@@ -0,0 +1,149 @@
+
+    public <T> UniqueTopicSerdeDecorator<T> decorateSerde(final Serde<T> delegate,
+                                                          final Properties config,
+                                                          final boolean isKey) {
+        final UniqueTopicSerdeDecorator<T> decorator = new UniqueTopicSerdeDecorator<>(delegate);
+        decorator.configure(config.entrySet().stream().collect(Collectors.toMap(e -> e.getKey().toString(), Map.Entry::getValue)), isKey);
+        return decorator;
+    }
+
+    public class UniqueTopicSerdeDecorator<T> implements Serde<T> {
+        private final AtomicBoolean isKey = new AtomicBoolean(false);
+        private final Serde<T> delegate;
+
+        public UniqueTopicSerdeDecorator(final Serde<T> delegate) {
+            this.delegate = delegate;
+        }
+
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            delegate.configure(configs, isKey);
+            this.isKey.set(isKey);
+        }
+
+        @Override
+        public void close() {
+            delegate.close();
+        }
+
+        @Override
+        public Serializer<T> serializer() {
+            return new UniqueTopicSerializerDecorator<>(isKey, delegate.serializer());
+        }
+
+        @Override
+        public Deserializer<T> deserializer() {
+            return new UniqueTopicDeserializerDecorator<>(isKey, delegate.deserializer());
+        }
+    }
+
+    public class UniqueTopicSerializerDecorator<T> implements Serializer<T> {
+        private final AtomicBoolean isKey;
+        private final Serializer<T> delegate;
+
+        public UniqueTopicSerializerDecorator(final AtomicBoolean isKey, final Serializer<T> delegate) {
+            this.isKey = isKey;
+            this.delegate = delegate;
+        }
+
+        @Override
+        public void configure(final Map<String, ?> configs, final boolean isKey) {
+            delegate.configure(configs, isKey);
+            this.isKey.set(isKey);
+        }
+
+        @Override
+        public byte[] serialize(final String topic, final T data) {
+            verifyTopic(topic, data);
+            return delegate.serialize(topic, data);
+        }
+
+        @Override
+        public byte[] serialize(final String topic, final Headers headers, final T data) {
+            verifyTopic(topic, data);
+            return delegate.serialize(topic, headers, data);
+        }
+
+        private void verifyTopic(final String topic, final T data) {
+            if (data != null) {
+                final String key = topic + (isKey.get() ? "--key" : "--value");
+                if (topicTypeRegistry.containsKey(key)) {
+                    assertThat(String.format("key[%s] data[%s][%s]", key, data, data.getClass()), topicTypeRegistry.get(key), equalTo(data.getClass()));
+                } else {
+                    System.out.printf("storing key[%s] data[%s][%s]%n", key, data, data.getClass());
```

Comment:
```suggestion
```

Reply:
Suggesting a commit to your own PR? seems they are the same :P

Reply:
Oops, that was just a self-suggestion to delete the line, which I forgot to actually click on.